url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/ScanFile.java

```diff
@@ -0,0 +1,343 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package backend;
+
+import backend.instructions.InsInterface3;
+import frontend.FrontEnd;
+
+import frontend.handlers;
+import java.awt.Color;
+import java.awt.Rectangle;
+import java.io.*;
+import java.util.StringTokenizer;
+import javax.swing.JEditorPane;
+import javax.swing.text.BadLocationException;
+
+/**
+ *
+ * @author nikhil
+ */
+public class ScanFile  implements Runnable 
+{
+
+   // public static ihandling ob;
+   public static boolean onconsole = false;
+    private static boolean commentflag = false;
+    public static boolean mainPresent;
+    public static int endOfProgram = -1;
+    public static long mainPosition;
+    Thread t;
+    public static File path;
+    SetBreakpoint ob_bkpt;
+//    public static boolean barrelshiftpresent = false;
+    public static int shiftType;
+    public static int reg1orimm0;
+    public static int regnoOrImmOp;
+    public static int shiftregno;
+    public static int ini_pos = 0;
+    public static Long pos = new Long(0);
+    public static RandomAccessFile br = null;
+    public static Integer current_line; //which line number is being executed;
+    static int ioflag = 0;
+    public static int call_count = 0;
+    public static String ins="";
+    public static String stepins="";
+    public static int exr=0;
+    public static String rawins="";
+
+    public static int getLineNumber(JEditorPane component, int pos)
+      {
+        int posLine;
+        int y = 0;
+
+        try
+        {
+            Rectangle caretCoords = component.modelToView(pos);
+            y = (int) caretCoords.getY();
+        } catch (BadLocationException ex)
+        {
+        }
+
+        int lineHeight = component.getFontMetrics(component.getFont()).getHeight();
+        posLine = (y / lineHeight) + 1;
+        return posLine;
+      }
+
+    
+    
+
+    public ScanFile(File p)
+      {
+
+        t = new Thread(this, "scan");
+        path = p;
+        t.start();
+      }
+
+    public ScanFile(File filepath, SetBreakpoint ob_bkpt)
+      {
+        t = new Thread(this, "scan");
+        path = filepath;
+        this.ob_bkpt = ob_bkpt;
+        t.start();
+      }
+
+    public void scanStart()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.